src/wire/ipv6ext.rs

```diff
@@ -0,0 +1,246 @@
+use {Error, Result};
+
+enum_with_unknown! {
+    /// IPv6 Extension Header Option type
+    pub doc enum OptionType(u8) {
```

Comment:
There should probably be a `OpFailureType` for the four on failure type defined in [RFC 8200 4.2]. Then a `From<OptionType> for OpFailureType`. E.g. something like

```
impl From<u8> for OpFailureType {
    fn from(other: u8) -> ExtOpFailureType {
        match other & 0b11000000 {
            0b00000000 => ExtOpFailureType::Skip,
            0b01000000 => ExtOpFailureType::Discard,
            0b10000000 => ExtOpFailureType::DiscardSendAll,
            0b11000000 => ExtOpFailureType::DiscardSendUnicast,
            _ => unreachable!()
        }
    }
}
```

[RFC 8200 4.2]: https://tools.ietf.org/html/rfc8200#section-4.2

Reply:
I did that at first, but removed it per:

> The three high-order bits described above are to be treated as part
   of the Option Type, not independent of the Option Type.  That is, a
   particular option is identified by a full 8-bit Option Type, not just
   the low-order 5 bits of an Option Type.

Thus, `0b00000005` and `0b00100005` were two distinct `OptionType`s.

Reply:
Yeah, you wouldn't change the current option type. I mean to add a new type that would be specifically for the following.

```
   The Option Type identifiers are internally encoded such that their
   highest-order 2 bits specify the action that must be taken if the
   processing IPv6 node does not recognize the Option Type:

      00 - skip over this option and continue processing the header.

      01 - discard the packet.

      10 - discard the packet and, regardless of whether or not the
           packet's Destination Address was a multicast address, send an
           ICMP Parameter Problem, Code 2, message to the packet's
           Source Address, pointing to the unrecognized Option Type.

      11 - discard the packet and, only if the packet's Destination
           Address was not a multicast address, send an ICMP Parameter
           Problem, Code 2, message to the packet's Source Address,
           pointing to the unrecognized Option Type.
```

This would essentially be used for something like

```rust
match packet.option_type() {
    ...
    OptionType::Unknown(type) {
        match OptFailureType::from(type) {
            OpFailureType::Skip => { skip the option }
            OpFailureType::Discard => { discard the packet }
            ...
        }
    }
}
```

I'm okay with tackling this in another issue and PR.

Reply:
I understand now., thank you. I will tackle that in a follow up PR.

Reply:
:+1: Sounds good